flycheck-ifort.el

```diff
@@ -67,29 +71,39 @@ for more information about warnings"
   :safe #'flycheck-string-list-p
   :package-version '(flycheck-ifort . "0.01"))
 
+(defun flycheck-ifort-parse-with-patterns (output checker buffer)
+  "Flycheck error parser for ifort. Converts lines which look like ------^ into a number."
+  (flycheck-parse-with-patterns
+   (replace-regexp-in-string "^-+\\^$"
+                             (lambda (s)
+                               (number-to-string (length s)))
+                             output)
+   checker buffer))
+
 (flycheck-define-checker fortran-ifort
   "An Fortran syntax checker using ifort.
 Uses Intel's Fortran compiler ifort.  See URL
 `https://software.intel.com/sites/default/files/m/f/8/5/8/0/6366-ifort.txt'."
-  :command ("~/.emacs.d/flycheck-ifort/ifortwrap"
+  :command ("ifort"
             "-syntax-only"
             (option "-stand" flycheck-ifort-language-standard)
             (option "" flycheck-ifort-layout concat
-                    flycheck-option-ifort-layout)
+                    flycheck-ifort-option-ifort-layout)
             (option-list "-warn" flycheck-ifort-warnings)
             (option-list "-I" flycheck-ifort-include-path concat)
             (eval flycheck-ifort-args)
             source)
+  :error-parser #'flycheck-ifort-parse-with-patterns
```

Comment:
Yeah, that makes sense and also seems to be the correct usage of `flycheck-define-checker`.

I think we assumed that `flycheck-define-checker` was a function, in which case functions as arguments would need to be passed as `#'foo` (or at least as `'foo`).

But it's a macro, and macros can break the rules. As a macro it could (and has) been implemented to take a function name without the `#'`.

Reply:
I'd seen the `'` used elsewhere in Flycheck while debugging, so was also confused as to why that didn't work here. It being a macro makes sense now; thanks for the explanation.